docs/development.md

```diff
@@ -1,54 +1,113 @@
+  - [Build Application](#build-application)
+  - [Build Runtime Server](#build-runtime-server)
+  - [Run Runtime Server](#run-runtime-server)
+  - [Run Example Client](#run-example-client)
+
+## Meta-Advice
+
+For any open-source project, the best way of figuring out what prerequisites and
+dependendencies are required is to look at the projects continuous integration
+(CI) scripts. A project's top-level [Travis](https://travis-ci.org/),
+[CircleCI](https://circleci.com/),
+[Cloudbuild](https://cloud.google.com/cloud-build) or
+[Dockerfile](https://docs.docker.com/) will give definitive information about
+what's needed, and at what version.
+
+For Oak, the CI system is currently Cloudbuild using a Docker image, so the
+top-level [`cloudbuild.yaml`](/cloudbuild.yaml) and [`Dockerfile`](Dockerfile)
+hold the project's configuration, dependencies and build scripts.
+
 ## Prerequisites
 
-We use Docker to install the base dependencies that need to exist at the system
-level, e.g. the [Rust compiler](https://www.rust-lang.org/tools/install) and the
-[Bazel build system](https://bazel.build); these steps are detailed in
-[`Dockerfile`](/Dockerfile). See
-https://docs.docker.com/engine/reference/builder/ for more information.
-
-Inside Docker, we use Bazel to version, install and build any non-Rust
-dependencies, including our own code. Dependencies are listed in
-[`WORKSPACE`](/WORKSPACE). See
-https://docs.bazel.build/versions/master/external.html for more information.
-
-To set up your development environment, you need the following applications. For
-the accurate versions required for a successful build please consult the current
-[`Dockerfile`](/Dockerfile).
-
-- **Docker**: follow
-  [Docker install instructions](https://docs.docker.com/install)
-- **Bazel**: follow
-  [Bazel install instructions](https://docs.bazel.build/versions/master/install.html)
-- **Rust**:
-  - Follow install instructions from https://rustup.rs/:
+The project's [`Dockerfile`](/Dockerfile) holds the definitive set of
+prerequisites for building and running Oak. The steps involved are generally
+clear from the contents, but the
+[Docker documentation](https://docs.docker.com/engine/reference/builder/) gives
+details for any steps that aren't obvious.
+
+The key prerequisites are:
+
+- **Docker**: If you want to use Docker to ensure an exact match with the
+  project's official build environment, follow the
+  [Docker install instructions](https://docs.docker.com/install).
+
+  If you want to work on the project without using Docker, you should
+  synchronize your installed versions of all tools with those specified in the
+  [`Dockerfile`](/Dockerfile).
+
+- **Rust**: The [Rust toolchain](https://www.rust-lang.org/tools/install) and
+  ancillary tools are required for building the project's source. Note that we
+  also [require](https://github.com/project-oak/oak/issues/969) a nightly build
+  of Rust; check the [`Dockerfile`](/Dockerfile) for the specific version.
```

Comment:
If I use docker, do I also need to install these? It seems like no?

Reply:
I've pulled out a separate "Docker vs. Native" section to try to make things a bit clearer.